compiler+runtime/src/cpp/jank/read/lex.cpp

```diff
@@ -524,15 +662,16 @@ namespace jank::read
                   return err(
                     error{ token_start, pos, "unexpected end of real, expecting exponent" });
                 }
-                break;
+                if(!is_letter(c) || (pos == token_start && file[token_start] != '0'))
+                {
+                  break;
+                }
```

Comment:
What're we checking here? Can you put whatever explanation you have into a comment, too, please? It's not obvious to me.

Reply:
added some comments for explanation.